.github/workflows/test-build.yml

```diff
@@ -98,56 +106,41 @@ jobs:
             type=ref,event=branch,priority=30
             type=sha,priority=40
 
-      - name: Build & push circ-exec image
-        uses: docker/build-push-action@v2
+      - name: Generate tags for circ-scripts image
+        id: scripts-tags
+        uses: docker/metadata-action@v3
         with:
-          context: .
-          file: ./docker/Dockerfile.exec
-          push: ${{ env.IMAGE_PUSH_ENABLED }}
-          tags: ${{ steps.exec-tags.outputs.tags }}
-
-
-  build-docker-scripts:
-    name: Build and push circ-scripts docker image
-    runs-on: ubuntu-latest
-    needs: [test]
-
-    # We want to run on external PRs, but not on our own internal PRs as they'll be run
-    # by the push to the branch. This prevents duplicated runs on internal PRs.
-    # Some discussion of this here:
-    # https://github.community/t/duplicate-checks-on-push-and-pull-request-simultaneous-event/18012
-    if: github.event_name == 'push' || github.event.pull_request.head.repo.full_name != github.repository
-
-    services:
-      postgres:
-        image: postgres:12
-        env:
-          POSTGRES_USER: ${{ env.POSTGRES_USER }}
-          POSTGRES_PASSWORD: ${{ env.POSTGRES_PASSWORD }}
-          POSTGRES_DB: ${{ env.POSTGRES_DB }}
+          images: ghcr.io/${{ github.repository_owner }}/circ-scripts
+          tags: |
+            type=semver,pattern={{major}}.{{minor}},priority=10
+            type=semver,pattern={{version}},priority=20
+            type=ref,event=branch,priority=30
+            type=sha,priority=40
 
-    steps:
-      - uses: actions/checkout@v2
+      - name: Generate tags for circ-webapp image
+        id: webapp-tags
+        uses: docker/metadata-action@v3
         with:
-          persist-credentials: false
-
-      - name: Set up Docker Buildx
-        uses: docker/setup-buildx-action@v1
+          images: ghcr.io/${{ github.repository_owner }}/circ-webapp
```

Comment:
Since these are only used when we are pushing images, can you move the tag generation for each image to right before we push the image?